### ISSUE TITLE
Ego vehicle drives through NPC vehicles

### ISSUE THREAD

**Issue by krtt** [lgsvl/simulator#1850]
Sim v2021.3 (Linux)
Apollo 6.0 (Modular testing)
Public map: Tartu Beta Release 3
Ego vehicle: SVL Lincoln (but similar problem is also with our custom car)

Illustrative video:

https://user-images.githubusercontent.com/22684441/139842278-76b70607-3dce-434d-80c5-1ad5cc8163ee.mp4

Related test scenario json created in VSE and run with VSE runner:

[vse_sim3_full_lap_fromPocket_lincoln_npcVehicles_bus.zip](https://github.com/lgsvl/simulator/files/7460431/vse_sim3_full_lap_fromPocket_lincoln_npcVehicles_bus.zip)

With NPC pedestrians it is the same or even worse where ego vehicle drives through them (but [this issue is already reported by somebody else](https://github.com/lgsvl/simulator/issues/1797)).

**Comment by EricBoiseLGSVL** [comment]
Yes, there is no collision in api simulations,  If you want that you need to change the source code.  See here. #1802
As for pedestrian ground truth, we are working on a fix for api spawned pedestrians and hope to release a fix soon.
